### DISCUSSION TITLE
[Kamal 2] Custom SSL/TLS certificate with kamal-proxy

### DISCUSSION THREAD

**Discussion by jessevdp** [basecamp/kamal#1049] [Q&A] [answered]
I noticed that `kamal-proxy` allows for a custom TLS certificate to be specified.

[kamal-proxy | Custom TLS certificate](https://github.com/basecamp/kamal-proxy?tab=readme-ov-file#custom-tls-certificate)

I does not seem like Kamal itself offers any way to configure this custom TLS certificate in its `proxy` configuration.

https://github.com/basecamp/kamal/blob/v2.1.1/lib/kamal/configuration/docs/proxy.yml

Am I missing something?

Wouldn't it be useful to have this option? Especially since the underlying kamal-proxy is already set up for it.

Or do people just install their custom certificates at a different location in the stack?

---

In my situation the infrastructure isn't quite setup in a way that meets al the prerequisites for Kamal's automatic SSL feature:

> This requires that we are deploying to one server and the host option is set. _**The host value must point to the server we are deploying to**_, and port 443 must be open for the Let's Encrypt challenge to succeed.

The server I'm deploying to sits behind some proxy server in what you could call a small private cloud. DNS points at this proxy server, not at my server directly. It seems like this is failing.

**Comment by Sija** [comment, marked as answer]
#969
